lib/workers/repository/update/pr/labels.ts

```diff
@@ -10,3 +12,69 @@ export function prepareLabels(config: RenovateConfig): string[] {
     .map((label) => template.compile(label, config))
     .filter(is.nonEmptyStringAndNotWhitespace);
 }
+
+/**
+ * Determine changed labels between old and new label arrays.
+ *
+ * This function takes two arrays of labels, 'oldLabels' and 'newLabels', and calculates the labels
+ * that need to be added and removed to transition from 'oldLabels' to 'newLabels'.
+ */
+export function getChangedLabels(
+  oldLabels: string[],
+  newLabels: string[],
+): Array<string[] | null> {
```

Comment:
```suggestion
):[string[] | undefined, string [] | undefined] {
```
you don't enforce null, so values can be undefined